test/ipfs.test.ts

```diff
@@ -0,0 +1,52 @@
+import { AddressZero } from "@ethersproject/constants";
+import * as chai from "chai";
+import { ethers } from "ethers";
+import { ThirdwebSDK, uploadMetadata } from "../src/index";
+import {readFileSync} from "fs";
+
+global.fetch = require("node-fetch");
+
+describe("NFT Module", async () => {
+  let sdk: ThirdwebSDK;
+
+  beforeEach(async () => {
+    sdk = new ThirdwebSDK(
+      new ethers.Wallet(
+        process.env.PKEY,
+        ethers.getDefaultProvider("https://rpc-mumbai.maticvigil.com"),
+      ),
+    );
+
+  });
+
+
+
+  it("should upload a file through any property, even when it is in an object nested inside another object", async () => {
+
+    try {
+      const upload = await uploadMetadata({
+        name: "test",
+        image: readFileSync("test/3510820011_4f558b6dea_b.jpg"),
+        test: {
+          test: {
+            image: readFileSync("test/3510820011_4f558b6dea_b.jpg")
+          }
+        }
+      })
+
+      chai.assert.isTrue(upload === "ipfs://bafkreidb4a4h3xg2ju6y3bmkp27ruottvf4gfmjccbtyj5hfk3uypftidu");
```

Comment:
For this test, rather than check the resulting IPFS hash thats returned, I want you to:

1. Upload (you already do this)
2. Download the uploaded data
3. Deserialize it and verify that the `image` and `test.test.image` properties both resolve to images in IPFS